searchcore/src/tests/proton/documentdb/lid_space_compaction/lid_space_common.cpp

```diff
@@ -0,0 +1,223 @@
+}
+IDocumentScanIterator::UP
+MyHandler::getIterator() const {
+    assert(_iteratorCnt < _lids.size());
+    return std::make_unique<MyScanIterator>(_lids[_iteratorCnt++], _bucketIdEqualLid);
+}
+MoveOperation::UP
+MyHandler::createMoveOperation(const search::DocumentMetaData &document, uint32_t moveToLid) const {
+    assert(document.lid > moveToLid);
+    _moveFromLid = document.lid;
+    _moveToLid = moveToLid;
+    return std::make_unique<MoveOperation>();
+}
+void
+MyHandler::handleMove(const MoveOperation &, IDestructorCallback::SP moveDoneCtx) {
+    ++_handleMoveCnt;
+    if (_storeMoveDoneContexts) {
+        _moveDoneContexts.push_back(std::move(moveDoneCtx));
+    }
+}
+void
+MyHandler::handleCompactLidSpace(const CompactLidSpaceOperation &op, std::shared_ptr<IDestructorCallback>) {
+    _wantedLidLimit = op.getLidLimit();
+}
+
+MyHandler::MyHandler(bool storeMoveDoneContexts, bool bucketIdEqualLid)
+    : _stats(),
+      _moveFromLid(0),
+      _moveToLid(0),
+      _handleMoveCnt(0),
+      _wantedLidLimit(0),
+      _iteratorCnt(0),
+      _storeMoveDoneContexts(storeMoveDoneContexts),
+      _bucketIdEqualLid(bucketIdEqualLid),
+      _moveDoneContexts(),
+      _op_listener(),
+      _rm_listener()
+{}
+
+MyHandler::~MyHandler() = default;
+
+#if 0
+struct MyStorer : public IOperationStorer {
+    uint32_t _moveCnt;
+    uint32_t _compactCnt;
+    MyStorer()
+        : _moveCnt(0),
+          _compactCnt(0)
+    {}
+    void appendOperation(const FeedOperation &op, DoneCallback) override {
+        if (op.getType() == FeedOperation::MOVE) {
+            ++ _moveCnt;
+        } else if (op.getType() == FeedOperation::COMPACT_LID_SPACE) {
+            ++_compactCnt;
+        }
+    }
+    CommitResult startCommit(DoneCallback) override {
+        return CommitResult();
+    }
+};
+
+struct MyFrozenBucketHandler : public IFrozenBucketHandler {
+    BucketId _bucket;
+    MyFrozenBucketHandler() : _bucket() {}
+    ExclusiveBucketGuard::UP acquireExclusiveBucket(BucketId bucket) override {
+        return (_bucket == bucket)
+               ? ExclusiveBucketGuard::UP()
+               : std::make_unique<ExclusiveBucketGuard>(bucket);
+    }
+    void addListener(IBucketFreezeListener *) override { }
+    void removeListener(IBucketFreezeListener *) override { }
+};
+
+struct MyFeedView : public test::DummyFeedView {
+    explicit MyFeedView(std::shared_ptr<const DocumentTypeRepo> repo)
+        : test::DummyFeedView(std::move(repo))
+    {
+    }
+};
+
+struct MyDocumentStore : public test::DummyDocumentStore {
+    Document::SP _readDoc;
+    mutable uint32_t _readLid;
+    MyDocumentStore() : _readDoc(), _readLid(0) {}
+    ~MyDocumentStore() override;
+    document::Document::UP read(search::DocumentIdT lid, const document::DocumentTypeRepo &) const override {
+        _readLid = lid;
+        return Document::UP(_readDoc->clone());
+    }
+};
+#endif
+
+MyDocumentStore::~MyDocumentStore() = default;
+
+#if 0
```

Comment:
Is this intentional? Consider removing the code block if not used.

Reply:
No, cleaned up.